### PR TITLE
Mobile Test 02 / 2018

### PR DESCRIPTION
**Pull request by ghost**
https://phabricator.wikimedia.org/T205108

There is still no text for the individual mini banner slides at this time which will have to be added later, the rest of the banner should be done though.

### CODE REVIEW

File: mobile_slider/banner_ctrl.js [hunk 1]
```diff
@@ -62,17 +61,7 @@ const weekdayPrepPhrase = dayName.isSpecialDayName() ? Translations[ 'day-name-p
 const $bannerContainer = $( '#WMDE-Banner-Container' );
 const CampaignName = $bannerContainer.data( 'campaign-tracking' );
 const BannerName = $bannerContainer.data( 'tracking' );
-const numberOfDaysUntilCampaignEnd = campaignDays.getNumberOfDaysUntilCampaignEnd();
```
Comment (tzhelyazkova) [inline mobile_slider/banner_ctrl.js:65]: Why is this removed?

File: mobile_slider/banner_ctrl.js [hunk 2]
```diff
@@ -91,9 +92,11 @@ $( '.select-group__option' ).click( function () {
 $( 'input[name=interval]' ).click( function () {
 	const subPaymentButton = $( 'button[data-payment-type=SUB]' );
 	if ( $( this ).attr( 'id' ) !== 'interval_onetime' ) {
+		if ( subPaymentButton.hasClass( 'selected-option' ) ) {
```
Comment (tzhelyazkova) [inline mobile_slider/banner_ctrl.js:92]: That check is probably not needed, because `removeClass()` won't return an error or do something unexpected when the class is not there.

File: mobile_slider/banner_ctrl.js [hunk 3]
```diff
@@ -118,6 +121,22 @@ $( '.payment-selection button' ).click( function ( event ) {
 	$( '#zahlweise' ).val( $( event.target ).data( 'payment-type' ) );
 } );
 
+$( '#banner-form-submit' ).click( function () {
+	if ( !$( '#periode' ).val() || jQuery( '.interval-selection .selected-option' ).length === 0 ) {
```
Comment (tzhelyazkova) [inline mobile_slider/banner_ctrl.js]: Is there a reason for using `jQuery` instead of `$` ?

**Review by tzhelyazkova — COMMENTED** [review]
I guess what's left is to split the code into var and ctrl parts.

File: mobile_slider/banner_ctrl.js (hunk 2 above)
Comment (ghost) [reply mobile_slider/banner_ctrl.js:92]: If it's not there, `$( '#zahlweise' ).val( '' );` will always reset the input field value when Sofortüberweisung is disabled. With the check in place, it will only do that if Sofortüberweisung was previously selected. I should probably add a comment that clarifies this though.

File: mobile_slider/banner_ctrl.js (hunk 3 above)
Comment (ghost) [reply mobile_slider/banner_ctrl.js]: Nah, I goofed after checking some stuff in the JavaScript console.

File: mobile_slider/banner_ctrl.js (hunk 1 above)
Comment (ghost) [reply mobile_slider/banner_ctrl.js:65]: It was only needed for the progress bar, it will be added back alongside the progress bar later during the campaign.

**Comment by tzhelyazkova** [comment]
Tried it on Safari, it looks pretty ok to me. 
Behaves as described in the ticket, besides the clicking on everywhere on the small banner to open the form. But that I assume would be impossible for now.

**Review by tzhelyazkova — APPROVED** [review]

**Comment by ghost** [comment]
@tzhelyazkova @gbirke Please check 3d362cb and the corresponding changes I have made to the BannerEvents:

https://meta.wikimedia.org/w/index.php?title=Schema:WMDEBannerEvents&oldid=18437830

**Comment by tzhelyazkova** [comment]
The changes seem alright to me :footprints:

**Comment by tzhelyazkova** [comment]
Merged this because it went live and the new `EventLogging` schema was also merged and deployed to production
